multigen/worker.py

```diff
@@ -32,14 +36,16 @@ def _update(sess, job, gs):
                     data['start_callback']()
 
                 pipe_name = sess.get('pipe', 'Prompt2ImPipe')
-                model_id = str(self.cwd/self.config["model_dir"]/self.models['base'][sess["model"]])
-                pipe = self.get_pipeline(pipe_name, model_id)
+                model_id = str(self.cwd/self.config["model_dir"]/self.models['base'][sess["model"]]['id'])
+                pipe = self.get_pipeline(pipe_name, model_id, cnet=data.get('cnet', None))
 
                 # TODO: the list of provided images can depend on the pipeline (e.g., it will be different for Cond)
                 images = data['images']
-                if images:
+                if len(images) == 2:

```

Comment:
condition should be pipeline id, not number of images

Reply:
I'm not sure to whom you address this comment since it is your code?